src/ray/core_worker/object_interface.h

```diff
@@ -0,0 +1,60 @@
+#ifndef RAY_CORE_WORKER_OBJECT_INTERFACE_H
+#define RAY_CORE_WORKER_OBJECT_INTERFACE_H
+
+#include "common.h"
+#include "ray/common/io.h"
+#include "ray/id.h"
+#include "ray/status.h"
+
+namespace ray {
+
+class CoreWorker;
+
+/// The interface that contains all `CoreWorker` methods that are related to object store.
+class CoreWorkerObjectInterface {
+ public:
+  CoreWorkerObjectInterface(CoreWorker *core_worker) { core_worker_ = core_worker; };
+
+  /// Put an object into object store.
+  ///
+  /// \param[in] buffer Data buffer of the object.
+  /// \param[out] object_id Generated ID of the object.
+  /// \return Status.
+  Status Put(const Buffer &buffer, const ObjectID *object_id);
+
+  /// Get a list of objects from the object store.
+  ///
+  /// \param[in] ids IDs of the objects to get.
+  /// \param[in] timeout_ms Timeout in milliseconds, wait infinitely if it's negative.
+  /// \param[out] results Result list of objects data.
+  /// \return Status.
+  Status Get(const std::vector<ObjectID> &ids, int64_t timeout_ms,
+             std::vector<Buffer> *results);
+
+  /// Wait for a list of objects to appear in the object store.
+  ///
+  /// \param[in] IDs of the objects to wait for.
+  /// \param[in] num_returns Number of objects that should appear.
+  /// \param[in] timeout_ms Timeout in milliseconds, wait infinitely if it's negative.
+  /// \param[out] results A bitset that indicates each object has appeared or not.
+  /// \return Status.
+  Status Wait(const std::vector<ObjectID> &object_ids, int num_objects,
+              int64_t timeout_ms, std::vector<bool> *results);
+
+  /// Delete a list of objects from the object store.
+  ///
+  /// \param[in] object_ids IDs of the objects to delete.
+  /// \param[in] local_only Whether only delete the objects in local node, or all nodes in
+  /// the cluster.
+  /// \param[in] delete_creating_tasks Whether also delete the tasks that
+  /// created these objects. \return Status.
+  Status Delete(const std::vector<ObjectID> &object_ids, bool local_only,
+                bool delete_creating_tasks);
+
+ private:
+  CoreWorker *core_worker_;
```

Comment:
This should be a shared_ptr to make sure memory management is done correctly

Reply:
Replied here https://github.com/ray-project/ray/pull/4875#discussion_r287889632